internal/pkg/admission/policy-server-configmap.go

```diff
@@ -34,38 +25,55 @@ type policyServerConfigEntry struct {
 func (r *Reconciler) reconcilePolicyServerConfigMap(
 	ctx context.Context,
 	policyServer *policiesv1alpha2.PolicyServer,
-	operation policyServerConfigMapOperation,
+	clusterAdmissionPolicies *policiesv1alpha2.ClusterAdmissionPolicyList,
 ) error {
+
 	cfg := &corev1.ConfigMap{}
 	err := r.Client.Get(ctx, client.ObjectKey{
 		Namespace: r.DeploymentsNamespace,
-		Name:      constants.PolicyServerConfigMapName,
+		Name:      policyServer.NameWithPrefix(),
 	}, cfg)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			if operation == RemovePolicy {
-				return nil
-			}
-			return r.createPolicyServerConfigMap(ctx, policyServer)
+			return r.createPolicyServerConfigMap(ctx, policyServer, clusterAdmissionPolicies)
 		}
 		return fmt.Errorf("cannot lookup policies ConfigMap: %w", err)
 	}
 
+	newPoliciesYML, err := r.createPoliciesYML(clusterAdmissionPolicies)
+	if err != nil {
+		return fmt.Errorf("cannot create policies: %w", err)
+	}
+	currentPoliciesYML := cfg.Data[constants.PolicyServerConfigPoliciesEntry]
+
+	if currentPoliciesYML != newPoliciesYML {
```

Comment:
I fear this could return false positives.

Let's say `currentPoliciesYML` is:
```yaml
policy1:
  url: registry.com/foo:1.2
  settings:
     name: flavio
     greeting: ciao
policy2:
  url: registry.com/bar:2
```


and `newPoliciesYML` is:
```yaml
policy2:
  url: registry.com/bar:2
policy1:
  url: registry.com/foo:1.2
  settings:
     name: flavio
     greeting: ciao
```
or something like:
```yaml
policy1:
  url: registry.com/foo:1.2
  settings:
     greeting: ciao
     name: flavio
policy2:
  url: registry.com/bar:2
```

This simple `!=` check will report the policy contents changed.

I think there's a high chance the ordering of these two structures is going to be completely messed up :(

Reply:
Maybe we could make your life easier by doing a change inside of policy-server...

Right now the `policy-server` ConfigMap has a single key called `policies.yml` which has a json string as value.
What if we change that so that:
  * The ConfigMap has multiple keys, one per `ClusterAdmissionPolicy`
  * The name of the key is going to be: `<ClusterAdmissionPolicy.name>.yml`
  * The value of the key is going to be a real Yaml document, that holds the information about the policy. Yaml can perserve the order of keys, which is what we care about

The Deployment will then mount the whole ConfigMap under something like `/policy-server/policies.d`.

The directory will look like that:
```
/policy-server/
     |
     ` policies.d/
           `- policy1.yml
           `- policy2.yml
```

The contents of `policy1,yml` would be:

```yaml
url: registry.com/foo:1.2
  settings:
     name: flavio
     greeting: ciao
```

The contents of `policy2.yml` would be:

```yaml
policy2:
  url: registry.com/bar:2
```

We would then change policy-server to have a new flag, something like `--policies-dir` which would load all the single files in there, merge them as a single Yaml and then do the usual stuff.

Just saying.... maybe this makes the Go comparison more reliable

Reply:
If that structure feels better I think we can go ahead, but I would not change the structure to fix something that is not very handy here.

I also think that we can check if comparing marshaled jsons work, go should alphabetically sort keys, so they should be the same regardless of the persisted order.

Reply:
you are right, I'll fix this

Reply:
we can parse the existing configmap into a map[string]policyServerConfigEntry, we are already creating a map[string]policyServerConfigEntry for the new policies. Then we can iterate and compare each key in the other map, and also compare the length. I'll write the code for that and I'll add unit tests to make sure it works in all scenarios.

Reply:
mm actually this will not work for settings. 
```
type policyServerConfigEntry struct {
	URL      string               `json:"url"`
	Settings runtime.RawExtension `json:"settings,omitempty"`
}
```
It will fail if the order in the settings is different

Reply:
Looking again I think I was right... With `encoding/json` we should be fine. I'm going to confirm but I think if we unmarshal, marshal and compare the marshaled objects we are good.

Reply:
Yes, check: https://pkg.go.dev/encoding/json

```
The map keys are sorted and used as JSON object keys by applying the following rules, subject to the UTF-8 coercion described for string values above:

- keys of any string type are used directly
- encoding.TextMarshalers are marshaled
- integer keys are converted to strings
```